routes/v1.js

```diff
@@ -502,52 +521,60 @@ api.declare({
     // otherwise the task would have been created with defineTask, and we don't
     // offer an idempotent operation in that case
     if (!_.isEqual(taskDef, def) || task.runs.length === 0) {
-      return res.reportError(
-        "RequestConflict", [
-          "taskId {{taskId}} already used by another task.",
-          "This could be the result of faulty idempotency!"
-        ].join('\n'),
-        {
-          taskId,
-        });
+      return res.reportError('RequestConflict', [
+        "taskId {{taskId}} already used by another task.",
+        "This could be the result of faulty idempotency!",
+        "Existing task definition was: {{existingTask}}",
+        "This request tried to define: {{taskDefinition}}",
+      ].join('\n'), {
+        taskId,
+        existingTask: def,
+        taskDefinition: taskDef,
+      });
     }
   }
 
-  // Construct task status, as we'll return this many times
-  var status = task.status();
+  // Track dependencies, if not already scheduled
+  if (task.state() === 'unscheduled') {
+    // Track dependencies, adds a pending run if ready to run
+    let err = await this.dependencyTracker.trackDependencies(task);
+    // We get an error here the task will be left in state = 'unscheduled',
+    // any attempt to use the same taskId will fail. And eventually the task
+    // will be resolved deadline-expired. But since createTask never returned
+    // successfully...
+    if (err) {
+      return res.reply('InputError', err.message, err.details);
+    }
+  }
 
-  // If first run isn't pending, all message must have been published before,
-  // this can happen if we came from the catch-branch (it's unlikely to happen)
-  if (task.runs[0].state !== 'pending') {
-    return res.reply({
-      status:   status
-    });
+  // Construct task status, as we'll return this many times
+  let status = task.status();
+
+  // If first run isn't unscheduled or pending, all message must have been
+  // published before, this can happen if we came from the catch-branch
+  // (it's unlikely to happen). But no need to publish messages again
+  let runZeroState = task.runs[0].state;
+  if (runZeroState !== 'unscheduled' && runZeroState !== 'pending') {
```

Comment:
We do this all over the place and it's not an issue with this line or this PR in particular, but I keep getting an itch to use something like an enum in these situations to avoid typo bugs. Maybe just a hash of states?
```js
let states = {unscheduled: 'unscheduled', pending: 'pending'};
// ...
if (runZeroState !== states.unscheduled && runZeroState !== states.pending) {
// ...
}
```
And now the part where you explain to me why this is a dumb idea can happen!